src/parser.js

```diff
@@ -380,7 +394,17 @@ function stateMachineStatementParser (isStrict, statement, steps) {
       }
 
       if (token.type === 'semicolon') {
-        statement.endStatement = ';';
+        // sqlite triggers have different syntax
+        if (dialect === 'sqlite' && (statement.type === 'CREATE_TRIGGER' && !statement.canEnd)) {
+          // do nothing
+        } else {
+          statement.endStatement = ';';
+          return;
+        }
```

Comment:
I think we can skip the empty if clause, and perhaps expand the above comment to include what about sqlite triggers is different (e.g. they include a semi-colon inside of their execute statement)

```suggestion
        if (dialect !== 'sqlite' || statement.type !== 'CREATE_TRIGGER' || statement.canEnd) {
          statement.endStatement = ';';
          return;
        }
```

Reply:
So I did do it the other way around, but from a code-reading perspective I like the flow of:

```

if <SQLITE>
  // do thing
If <SQLSERVER
  // do thing
```

rather than if not [set of databases]. Feel like it makes it more explicit.